src/controllers/socialAuthController.js

```diff
@@ -0,0 +1,135 @@
+import crypto from 'crypto';
+import { generateToken } from '../middlewares/authentication';
+import models from '../db/models';
+
+const { User } = models;
+
+/**
+ * @description Social authentication controller
+ * @class SocialAuthController
+ */
+class SocialAuthController {
```

Comment:
Nice work Kizito, but however I think we could change the class name to SocilaAuthentication without the adding the controller part.